lxd/container_lxc.go

```diff
@@ -2519,22 +2519,24 @@ func (c *containerLXC) startCommon() (string, []func() error, error) {
 			if m["path"] != "/" {
 				diskDevices[k] = m
 			}
-		} else if m["type"] == "nic" || m["type"] == "infiniband" || m["type"] == "proxy" {
+		} else if shared.StringInSlice(m["type"], []string{"nic", "proxy", "infiniband"}) {
 			networkKeyPrefix := "lxc.net"
 			if !util.RuntimeLiblxcVersionAtLeast(2, 1, 0) {
 				networkKeyPrefix = "lxc.network"
 			}
 
 			// Populate network nic name and hwaddr.
-			if m["type"] == "nic" || m["type"] == "infiniband" {
+			if shared.StringInSlice(m["type"], []string{"nic", "infiniband"}) {
+				// Increment nicID so that LXC network index is unique per device.
+				nicID++
```

Comment:
Ah, looks like you're fixing this issue here now. So you probably want to do some rebasing to make those commits look better on their own :)

Reply:
@stgraber I've moved the nicID change into the former commit so that it sits together with the re-organisation of logic to only do fillNetworkDevice if the device is nic or infiniband (which is there to accomodate the addition of the postStartHook).